src/NUnitEngine/nunit-agent/AgentExitCodes.cs

```diff
@@ -0,0 +1,36 @@
+﻿// ***********************************************************************
+// Copyright (c) 2017 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+namespace NUnit.Common
+{
+    internal static class AgentExitCodes
+    {
+        public const int OK = 0;
+        public const int PARENT_PROCESS_TERMINATED = -1;
+        public const int UNEXPECTED_EXCEPTION = -2;
```

Comment:
It's a nit, but maybe this should be -100 as for the console runner. That would allow us to eventually consolidate the return codes.

Reply:
This one annoyed me - I wasn't quite sure what to call it. It's not really equivalent to the console's -100 'unexpected exception anywhere' - as this code only wraps a specific part of the agent (post-initialisation.). That was about as much info as was in the original log message though, and I wasn't sure what else to call it!

I don't mind changing it to -100 - would you envisage these ever being consolidated though? The list of reasons doesn't seem to have much in common.

Reply:
In the console UNEXPECTED_EXCEPTION generally means it's our error, even if it's due to a user error we neglected to handle. No user exceptions are "expected" to escape. I think it's the same here.

Yes, I think this needs a better name but I don't know what it might be.